### ISSUE TITLE
FrameConvexPolygon2D (and likely other classes) can't handle out of order vertices

### ISSUE THREAD

**Issue by rjgriffin42** [ihmcrobotics/euclid#17]
The following test passes:

```Random random = new Random(1738L);
Point2D pointA = EuclidCoreRandomTools.nextPoint2D(random, 10.0);
Point2D pointB = EuclidCoreRandomTools.nextPoint2D(random, 10.0);

FrameConvexPolygon2D polygon = new FrameConvexPolygon2D();
polygon.addVertex(pointA);
polygon.addVertex(pointA);
polygon.addVertex(pointB);
polygon.addVertex(pointB);

polygon.update();

assertEquals(2, polygon.getNumberOfVertices());
```

while the following test fails:

```Random random = new Random(1738L);
Point2D pointA = EuclidCoreRandomTools.nextPoint2D(random, 10.0);
Point2D pointB = EuclidCoreRandomTools.nextPoint2D(random, 10.0);

FrameConvexPolygon2D polygon = new FrameConvexPolygon2D();
polygon.addVertex(pointA);
polygon.addVertex(pointB);
polygon.addVertex(pointA);
polygon.addVertex(pointB);

polygon.update();

assertEquals(2, polygon.getNumberOfVertices());
```

Something in the giftwrapping of the vertices is not functioning properly.

**Comment by rjgriffin42** [comment]
This test additional fails, which should pass:

```Random random = new Random(3245345L);

      for (int i = 0; i < ITERATIONS; i++)
      {
         Point2D pointA = EuclidCoreRandomTools.nextPoint2D(random, 10.0);
         Point2D pointB = EuclidCoreRandomTools.nextPoint2D(random, 10.0);
         List<Point2D> points = new ArrayList<>();
         points.add(pointA);
         points.add(pointB);
         points.add(pointA);
         points.add(pointB);



         int expectedHullSize = EuclidGeometryPolygonTools.inPlaceGiftWrapConvexHull2D(points, 4);
         assertEquals(2, expectedHullSize);
      }
```

**Comment by SylvainBertrand** [comment]
Ok thanks for the test, I'll work on a fix.

**Comment by SylvainBertrand** [comment]
Addressed in ed4cc900f5ef121370cae3494b8dd2dd71657224
